base/src/main/kotlin/com/freeletics/gradle/plugin/FreeleticsBasePlugin.kt

```diff
@@ -76,20 +75,19 @@ public abstract class FreeleticsBasePlugin : Plugin<Project> {
                 toolchain.vendor.set(JvmVendorSpec.AZUL)
             }
 
-            compilerOptions(project) {
+            compilerOptions {
                 getVersionOrNull("kotlin-language")?.let {
                     languageVersion.set(KotlinVersion.fromVersion(it))
                 }
 
                 allWarningsAsErrors.set(!booleanProperty("fgp.kotlin.allowWarnings", false).get())
 
-                freeCompilerArgs.addAll(
-                    // In this mode, some deprecations and bug-fixes for unstable code take effect immediately.
-                    "-progressive",
-                    // Support inferring type arguments based on only self upper bounds of the corresponding type parameters
-                    // https://kotlinlang.org/docs/whatsnew1530.html#improvements-to-type-inference-for-recursive-generic-types
-                    "-Xself-upper-bound-inference",
-                )
+                // In this mode, some deprecations and bug-fixes for unstable code take effect immediately.
+                progressiveMode.set(true)
+
+                // Support inferring type arguments based on only self upper bounds of the corresponding type parameters
+                // https://kotlinlang.org/docs/whatsnew1530.html#improvements-to-type-inference-for-recursive-generic-types
+                freeCompilerArgs.add("-Xself-upper-bound-inference")
```

Comment:
Same here for progressive mode https://kotlinlang.org/docs/whatsnew19.html#new-compiler-options